std/conv.d

```diff
@@ -5338,3 +5338,229 @@ unittest
+    static assert( !"12gG"w.isHexLitteral);
+    static assert( !"2A  3q"w.isHexLitteral);
+    static assert( !"12gG"d.isHexLitteral);
+    static assert( !"2A  3q"d.isHexLitteral);
+    // valid x strings
+    static assert( ("5A" ~ whitespace).isHexLitteral);
+    static assert( ("5A 01A C FF de 1b").isHexLitteral);
+    static assert( ("0123456789abcdefABCDEF").isHexLitteral);
+    static assert( (" 012 34 5 6789 abcd ef\rAB\nCDEF").isHexLitteral);
+    static assert( ("5A 01A C FF de 1b"w).isHexLitteral);
+    static assert( ("0123456789abcdefABCDEF"w).isHexLitteral);
+    static assert( (" 012 34 5 6789 abcd ef\rAB\nCDEF"w).isHexLitteral);
+    static assert( ("5A 01A C FF de 1b"d).isHexLitteral);
+    static assert( ("0123456789abcdefABCDEF"d).isHexLitteral);
+    static assert( (" 012 34 5 6789 abcd ef\rAB\nCDEF"d).isHexLitteral);
+    // library version allows what's pointed by issue 10454
+    static assert( ("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF").isHexLitteral);
+}
+
+/**
+The $(D hexString) function is intended to replace the hexadecimal literal strings
+starting with $(D 'x'), which could be removed to simplify the core language.
+
+The function takes a string made of hexadecimal digits and it returns
+the matching $(D string) by converting each pair of digit to a character.
+The input string can also include some ASCII white characters, which can be used
+to keep the literal string readable in the source code.
+
+Params:
+hexData = the $(D string) to be converted.
+C = character type of the output string. By default the function returns a string
+but can also return a $(D wstring) or a $(D dstring) if the parameter is set to
+$(D wchar) or $(D dchar).
+ */
+@property @safe nothrow pure
+auto hexString(string hexData, C = char)()
+if (hexData.isHexLitteral && isSomeChar!C)
+{
+    return hexStrImpl!(C)(hexData);
+}
+
+/// Ditto
+@property @safe pure
+auto hexString(C = char)(string hexData)
+if (isSomeChar!C)
+{
+    if (hexData.isHexLitteral)
+        return hexStrImpl!(C)(hexData);
+    else
+        assert(0, "Invalid input string format in " ~ __FUNCTION__);
+}
+
+///
+unittest
+{
+    // conversion at compile time
+    auto string1 = hexString!"304A314B";
+    assert(string1 == "0J1K");
+    // conversion at run time
+    auto arbitraryData = "30 4A 31 4B";
+    auto string2 = hexString(arbitraryData);
+    assert(string2 == "0J1K");
+}
+
+/**
+The $(D hexBytes) function is similar to the $(D hexString) one except
+that it returns the hexadecimal data as an array of integral.
+
+Params:
+hexData = the $(D string) to be converted.
+T = the integer type of an array element. By default it is set to $(D ubyte) 
+but all the integer type, excepted $(D ulong) and $(D long), are accepted. 
+ */
+@property @trusted nothrow pure
+auto hexBytes(string hexData, T = ubyte)()
+if (hexData.isHexLitteral && isIntegral!T && (T.sizeof <= 4))
+{
+    static if (T.sizeof == 1)
+        return cast(T[]) hexStrImpl!char(hexData);
```

Comment:
Casting from code units to integers really shouldn't affect `@safe`, grumble grumble...